includes/WikiDiscover.php

```diff
@@ -117,111 +108,159 @@ public function isInactive( $database ) {
 	public function isPrivate( $database ) {
 		return in_array( $database, $this->private );
 	}
+	
+	/**
+	 * @param Parser $parser
+	 */
+	public static function onParserFirstCallInit( Parser $parser ) {
+		$parser->setFunctionHook( 'numberofwikisincategory', [ __CLASS__, 'numberOfWikisInCategory' ], Parser::SFH_NO_HASH );
+		$parser->setFunctionHook( 'numberofwikisinlanguage', [ __CLASS__, 'numberOfWikisInLanguage' ], Parser::SFH_NO_HASH );
+		$parser->setFunctionHook( 'numberofwikisbysetting', [ __CLASS__, 'numberOfWikisBySetting' ], Parser::SFH_NO_HASH );
+	}
+	
+	/**
+	 * @param Parser $parser
+	 * @param String $category|null
+	 * @return integer
+	 */
+	public static function numberOfWikisInCategory( Parser $parser, String $category = null ) {
+		$config = MediaWikiServices::getInstance()->getMainConfig();
+
+		$dbw = wfGetDB( DB_MASTER, [], $config->get( 'CreateWikiDatabase' ) );
+
+		return $dbw->selectRowCount( 'cw_wikis', '*', [ 'wiki_deleted' => 0, 'wiki_category' => strtolower( $category ) ] );;
+	}
+
+	/**
+	 * @param Parser $parser
+	 * @param String $language|null
+	 * @return integer
+	 */
+	public static function numberOfWikisInLanguage( Parser $parser, String $language = null ) {
+		$config = MediaWikiServices::getInstance()->getMainConfig();
+
+		$dbw = wfGetDB( DB_MASTER, [], $config->get( 'CreateWikiDatabase' ) );
 
+		return $dbw->selectRowCount( 'cw_wikis', '*', [ 'wiki_deleted' => 0, 'wiki_language' => strtolower( $language ) ] );;
+	}
+	
+	/**
+	 * @param Parser $parser
+	 * @param mixed $setting|null
+	 * @param mixed $value|null
+	 * @return string|integer
+	 */
+	public static function numberOfWikisBySetting( $parser, $setting = null, $value = null ) {
+		if ( !$setting && !$value ) {
```

Comment:
Are both required, or is only one required?

Reply:
Only one of them is required.